nextcloud/terraform/nextcloud.tf

```diff
@@ -0,0 +1,218 @@
+# Terraform config file
+
+terraform {
+  required_providers {
+    openstack = {
+      source = "terraform-provider-openstack/openstack"
+    }
+  }
+}
+
+provider "openstack" {
+  allow_reauth = false
+}
+
+# Parameters/Variables
+
+variable "host_name" {
+  description = "host name to associate with IP address"
+  type = string
+  default = <your-host-name>
+}
+
+variable "domain_name" {
+  description = "domain name to associate with IP address"
+  type = string
+  default = <your-domain-name>
+}
+
+variable "ddns_script_url" {
+  description = "URL of a script that will configure update ddns
+  (called as ./ddns-script.sh <hostname> <ip> <password>)"
+  type = string
+  default = "https://raw.githubusercontent.com/flashvoid/demo-provision/main
```

Comment:
1) `ddns-update` now lives at https://raw.githubusercontent.com/yvonnewat/catalystcloud-orchestration/new/nextcloud-tutorial-resources/tools/ddns/namecheap/ddns-update.sh
2) After successful merge it will be under https://raw.githubusercontent.com/catalyst-cloud/catalystcloud-orchestration/master/tools/ddns/namecheap/ddns-update.sh
3) put the number 1 url now and right after merge we will fix this to be url number 2